crates/task-impls/src/vote_collection.rs

```diff
@@ -105,36 +104,6 @@ impl<
     }
 }
 
-impl<
-        TYPES: NodeType,
-        VOTE: Vote<TYPES>
-            + AggregatableVote<TYPES, VOTE, CERT>
-            + std::marker::Send
-            + std::marker::Sync
-            + 'static,
-        CERT: Certificate<TYPES, Voteable = VOTE::Commitment>
-            + Debug
-            + std::marker::Send
-            + std::marker::Sync
-            + 'static,
-    > TaskState for VoteCollectionTaskState<TYPES, VOTE, CERT>
-where
-    VoteCollectionTaskState<TYPES, VOTE, CERT>: HandleVoteEvent<TYPES, VOTE, CERT>,
-{
-    type Event = Arc<HotShotEvent<TYPES>>;
-
-    type Output = HotShotTaskCompleted;
-
-    async fn handle_event(event: Self::Event, task: &mut Task<Self>) -> Option<Self::Output> {
-        let sender = task.clone_sender();
-        task.state_mut().handle_event(event, &sender).await
-    }
-
-    fn should_shutdown(event: &Self::Event) -> bool {
-        matches!(event.as_ref(), HotShotEvent::Shutdown)
-    }
-}
-
```

Comment:
I think implementing `TaskState` here (and calling the function `handle_event` below) was misleading, since we were never spawning the vote task as a task.

This may be a `CompletableTaskState` in the future, but that is outside the scope of this PR.